File: backend/app/schemas/encargado.py
from pydantic import BaseModel, ConfigDict, EmailStr, Field

from app.schemas.usuario import UsuarioOut


class EstudianteMini(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id_estudiante: int
    name_estudiante: str
    sec_name_estudiante: str


class EncargadoBase(BaseModel):
    name_encargado: str = Field(min_length=1, max_length=100)
    sec_name_encargado: str = Field(min_length=1, max_length=100)
    id_tipo_documento: int
    num_documento_encargado: str = Field(min_length=1, max_length=30)
    phone_num_encargado: str | None = Field(default=None, max_length=20)
    direction_encargado: str | None = Field(default=None, max_length=255)
    parentesco: str = Field(min_length=1, max_length=50)


class EncargadoCreate(EncargadoBase):
    correo_institucional: EmailStr
    password: str = Field(min_length=8)
    estudiantes_ids: list[int] = Field(default_factory=list)


class EncargadoUpdate(BaseModel):
    name_encargado: str | None = Field(default=None, min_length=1, max_length=100)
    sec_name_encargado: str | None = Field(default=None, min_length=1, max_length=100)
    id_tipo_documento: int | None = None
    num_documento_encargado: str | None = Field(default=None, min_length=1, max_length=30)
    phone_num_encargado: str | None = Field(default=None, max_length=20)
    direction_encargado: str | None = Field(default=None, max_length=255)
    parentesco: str | None = Field(default=None, min_length=1, max_length=50)
    estudiantes_ids: list[int] | None = None


class EncargadoOut(EncargadoBase):
    model_config = ConfigDict(from_attributes=True)

    id_encargado: int
    usuario: UsuarioOut
    estudiantes: list[EstudianteMini]

File: tailwind-admin-reactjs-free/package/src/types/encargado.ts
import type { Usuario } from 'src/types/estudiante';

export interface EstudianteMini {
  id_estudiante: number;
  name_estudiante: string;
  sec_name_estudiante: string;
}

export interface Encargado {
  id_encargado: number;
  name_encargado: string;
  sec_name_encargado: string;
  id_tipo_documento: number;
  num_documento_encargado: string;
  phone_num_encargado: string | null;
  direction_encargado: string | null;
  parentesco: string;
  usuario: Usuario;
  estudiantes: EstudianteMini[];
}

export interface EncargadoCreate {
  correo_institucional: string;
  password: string;
  name_encargado: string;
  sec_name_encargado: string;
  id_tipo_documento: number;
  num_documento_encargado: string;
  phone_num_encargado?: string | null;
  direction_encargado?: string | null;
  parentesco: string;
  estudiantes_ids: number[];
}

export interface EncargadoUpdate {
  name_encargado: string;
  sec_name_encargado: string;
  id_tipo_documento: number;
  num_documento_encargado: string;
  phone_num_encargado?: string | null;
  direction_encargado?: string | null;
  parentesco: string;
  estudiantes_ids: number[];
}
